osu.Game.Tournament/Models/TournamentMatch.cs

```diff
@@ -106,13 +106,16 @@ public void CancelMatchStart()
         }
 
         /// <summary>
-        /// Initialise this match with zeroed scores. Will be a noop if either team is not present.
+        /// Initialise this match with zeroed scores. Will be a noop if either team is not present or if either of the scores are non-zero.
         /// </summary>
         public void StartMatch()
         {
             if (Team1.Value == null || Team2.Value == null)
                 return;
 
+            if (Team1Score.Value > 0 || Team2Score.Value > 0)
+                return;
+
             Team1Score.Value = 0;
             Team2Score.Value = 0;
```

Comment:
In the tournament client, a match with null scores is considered "not started". This is used for display purposes in the bracket screen (and maybe the schedule screen?), with scores not showing (rather than 0-0 after starting a match).

So yes, this method still initialises the scores to non-null and marks the match as "currently in progress", if that's what you mean.